src/ProgramBuilder.ts

```diff
@@ -498,5 +510,8 @@ export class ProgramBuilder {
         if (this.program) {
             this.program.dispose?.();
         }
+        if (this.watchInterval) {
+            clearInterval(this.watchInterval);
```

Comment:
Calling dispose while running a watcher-enabled builder will now allow the process terminate successfully.